File: app/src/main/java/com/example/visiba/visibarectest/LanguageHandler.java
package com.example.visiba.visibarectest;

import android.content.Context;
import android.content.res.Configuration;

import java.util.Locale;

public class LanguageHandler
{
    public static void setLocale(String lang, Context context) {
        Locale locale = new Locale(lang);
        Locale.setDefault(locale);

        Configuration configuration = context.getResources().getConfiguration();
        configuration.setLocale(locale);
        context.getResources().updateConfiguration(configuration, null);
    }
}

File: app/src/main/java/com/example/visiba/visibarectest/LanguageHandler.kt
package com.example.visiba.visibarectest

import android.content.Context
import android.content.res.Configuration

import java.util.Locale

object LanguageHandler {
    fun setLocale(lang: String, context: Context) {
        val locale = Locale(lang)
        Locale.setDefault(locale)

        val configuration = context.resources.configuration
        configuration.setLocale(locale)
        context.resources.updateConfiguration(configuration, null)
    }
}
